controllers/k8ssandra/datacenters.go

```diff
@@ -273,23 +281,14 @@ func (r *K8ssandraClusterReconciler) setStatusForDatacenter(kc *api.K8ssandraClu
 	}
 }
 
-func (r *K8ssandraClusterReconciler) removeDatacenterStatus(kc *api.K8ssandraCluster, dcName string) {
-	if kdcStatus, found := kc.Status.Datacenters[dcName]; found {
-		if kdcStatus.Stargate == nil {
-			delete(kc.Status.Datacenters, dcName)
-		} else {
-			kc.Status.Datacenters[dcName] = api.K8ssandraStatus{
-				Stargate:  kdcStatus.Stargate.DeepCopy(),
-				Cassandra: nil,
-				Reaper:    kdcStatus.Reaper.DeepCopy(),
-			}
-		}
-	}
-}
-
 func datacenterAddedToExistingCluster(kc *api.K8ssandraCluster, dcName string) bool {
 	_, found := kc.Status.Datacenters[dcName]
-	return kc.Status.GetConditionStatus(api.CassandraInitialized) == corev1.ConditionTrue && !found
+	if kc.Spec.Cassandra.ServerType == api.DistributionDse {
```

Comment:
Do we have a different logic for DSE here since we cannot apply system replication settings like we do for Cassandra?

Reply:
Yes. For Cassandra, we'll only perform alter+rebuild in the case where we're not initializing the cluster, even if there are multiple DCs being created.
For DSE we need to perform these operations for all DCs except the first one, even if the cluster isn't fully initialized.